dxr/static_unhashed/js/dxr.js

```diff
@@ -411,7 +411,13 @@ $(function() {
                     if (addToHistory) {
                         var pushHistory = function () {
                             // Strip off offset= and limit= when updating.
-                            var displayURL = queryString.replace(/[&?]offset=\d+/, '').replace(/[&?]limit=\d+/, '');
+                            function dropAmp(fullMatch, ampOrQuestion) {
+                                // Drop a leading ampersand, keep a leading question mark.
+                                return (ampOrQuestion === '?') ? '?' : '';
+                            }
+                            var displayURL = queryString.replace(/([&?])offset=\d+/, dropAmp).
+                                                         replace(/([&?])limit=\d+/, dropAmp).
+                                                         replace('?&', '?');
```

Comment:
On second thought, maybe this is better:
```diff
- // Strip off offset= and limit= when updating.
- var displayURL = queryString.replace(/[&?]offset=\d+/, '').replace(/[&?]limit=\d+/, '');
+ // Strip off offset= and limit= when updating - leave a '?', drop an '&'.
+ var displayURL = queryString.replace(/(?:(\?)|&)offset=\d+/, '$1').
+                              replace(/(?:(\?)|&)limit=\d+/, '$1').
+                              replace('?&', '?');
```
It's a little harder to read (for me at least), but more professional-looking and concise I guess :P  Any preference?